### ISSUE TITLE
autogen.sh not working correctly

### ISSUE THREAD

**Issue by SevenITGithub** [horst3180/Vertex-theme#93]
https://github.com/horst3180/Vertex-theme/blob/master/autogen.sh

user@arandompc:~/gitprojects/vertex-theme$ ./autogen.sh --prefix=/usr
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /usr/bin/autoconf --force
configure.ac:11: error: possibly undefined macro: AC_CONFIG_MACRO_DIRS
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /usr/bin/autoconf failed with exit status: 1

I'm running Debian Wheezy and I think I followed the instructions correctly. I installed all the required packages correctly too.

**Comment by horst3180** [comment]
Looks like Wheezy's version of automake/autoconf is too old. Could you try changing

    AC_CONFIG_MACRO_DIRS([m4])

to

    AC_CONFIG_MACRO_DIR([m4])

in `configure.ac` and see if that fixes it?

Also Wheezy's gtk3 version is not supported by Vertex, this will probably cause another error when you attempt to build it. You could try `./autogen.sh --prefix=/usr --with-gnome=3.10` to build the theme for the oldest supported gtk3 version.

**Comment by SevenITGithub** [comment]
Thank you for answering. I changed configure.ac and ran like you said:

user@arandompc:~/gitprojects/vertex-theme$ ./autogen.sh --prefix=/usr --with-gnome=3.10
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /usr/bin/autoconf --force
autoreconf: configure.ac: not using Autoheader
autoreconf: running: automake --add-missing --copy --force-missing
common/Makefile.am:13: ENABLE_DARK does not appear in AM_CONDITIONAL
common/Makefile.am:20: ENABLE_LIGHT does not appear in AM_CONDITIONAL
common/Makefile.am:27: ENABLE_CINNAMON does not appear in AM_CONDITIONAL
common/Makefile.am:33: ENABLE_GNOME_SHELL does not appear in AM_CONDITIONAL
common/Makefile.am:39: ENABLE_GTK2 does not appear in AM_CONDITIONAL
common/Makefile.am:43: ENABLE_DARK does not appear in AM_CONDITIONAL
common/Makefile.am:47: ENABLE_LIGHT does not appear in AM_CONDITIONAL
common/Makefile.am:55: ENABLE_GTK3 does not appear in AM_CONDITIONAL
common/Makefile.am:62: ENABLE_DARK does not appear in AM_CONDITIONAL
common/Makefile.am:73: ENABLE_LIGHT does not appear in AM_CONDITIONAL
common/Makefile.am:85: ENABLE_METACITY does not appear in AM_CONDITIONAL
common/Makefile.am:96: ENABLE_DARK does not appear in AM_CONDITIONAL
common/Makefile.am:113: ENABLE_LIGHT does not appear in AM_CONDITIONAL
common/Makefile.am:133: ENABLE_UNITY does not appear in AM_CONDITIONAL
common/Makefile.am:136: ENABLE_DARK does not appear in AM_CONDITIONAL
common/Makefile.am:140: ENABLE_LIGHT does not appear in AM_CONDITIONAL
common/Makefile.am:147: ENABLE_XFWM does not appear in AM_CONDITIONAL
common/Makefile.am:150: ENABLE_DARK does not appear in AM_CONDITIONAL
common/Makefile.am:154: ENABLE_LIGHT does not appear in AM_CONDITIONAL
common/Makefile.am:164: ENABLE_DARK does not appear in AM_CONDITIONAL
common/Makefile.am:168: ENABLE_LIGHT does not appear in AM_CONDITIONAL
autoreconf: automake failed with exit status: 1

**Comment by horst3180** [comment]
@jgeboski Any idea?

@SevenITGithub You could probably get away with installing the Ubuntu 14.04 package until this is fixed. I haven't tested this, though.

**Comment by horst3180** [comment]
Just to warn you. It's probably not a good idea to install this theme on Wheezy if you plan to use Gtk3 applications or even Gnome. The Gtk3 and Gnome-Shell themes will most likely look completely broken due to the old Gtk/Gnome-Shell versions.

**Comment by SevenITGithub** [comment]
It's not that important for me to have this theme, I just like it. Right now I have Delorean Light and it looks okay. I just don't want to break Debian. Thank you very much for your help.

automake: 1.11.6
autoconf: 2.69

http://gnome-look.org/content/show.php/DeLorean-Light-Theme-3.6?content=156144

**Comment by jgeboski** [comment]
It would appear things are broken with the `VERTEX_ENABLE()` macro on older versions of autoconf.  I will have to look at the issue more in dept to say for sure.

Also, we should migrate from `AC_CONFIG_MACRO_DIRS()` to `AC_CONFIG_MACRO_DIR()`.  I meant the second one anyways.  I have no idea why I ended up appending the `S`.

**Comment by horst3180** [comment]
I found the culprit. `AC_CONFIG_MACRO_DIR` was added to automake in version 1.13. I could make it build with the following adjustments:

    ---
     Makefile.am  |    2 ++
     configure.ac |    1 -
     2 files changed, 2 insertions(+), 1 deletion(-)

    diff --git a/Makefile.am b/Makefile.am
    index 140e276..0cf86c1 100644
    --- a/Makefile.am
    +++ b/Makefile.am
    @@ -1,2 +1,4 @@
    +ACLOCAL_AMFLAGS = -I m4
    +
     EXTRA_DIST = autogen.sh
     SUBDIRS    = common extra
    diff --git a/configure.ac b/configure.ac
    index 341f338..d8ac54b 100644
    --- a/configure.ac
    +++ b/configure.ac
    @@ -8,7 +8,6 @@ AC_INIT(
     )
 
     AC_CONFIG_AUX_DIR([build-aux])
    -AC_CONFIG_MACRO_DIR([m4])
     AM_INIT_AUTOMAKE([foreign])
     AC_PROG_MKDIR_P
 
    -- 

However, it turns out that the gtk2 theme is broken, too under Debian 7. That's because it requires the Adwaita engine, which is not included/didn't exist when Debian 7 was released. So I don't know if we even need that patch. The theme won't work on those old systems anyway.
My decision is not to support Debian 7 and older. Debian 8 is coming soon though and everything should work just fine there.

**Comment by jgeboski** [comment]
I would leave `AC_CONFIG_MACRO_DIR([m4])` in `configure.ac`, and just add `ACLOCAL_AMFLAGS = -Im4` to `Makefile.am`.  I know libtool throws a warning when `-Im4` is not in `ACLOCAL_AMFLAGS`, but this project does not use libtool, so I left it out.

Also, checks for the GTK engines could probably be added.  It would be really simple if they drop a `pc` file in place.

**Comment by horst3180** [comment]
Sure, we could add some additional checks, but in this case it won't work. The Adwaita engine is a part of the `gnome-themes-standard` package, but it was created after the release of Debian 7, so it's not included. Also it seems that `gnome-themes-standard` doesn't provide a `.pc` file.
I could probably remove the Adwaita dependency from the gtk2 theme. I just have to find out what exactly the engine does and if it could be replaced.

**Comment by horst3180** [comment]
Ok, I got the Adwaita dependency removed without major drawbacks. Now there's another problem. Wheezy uses `gtk+ 2.24.10`, but in `gtk+ 2.24.11` an option to select for an LTR/RTL widget direction was added to the pixbuf engine. Of course the older gtk version doesn't recognize this and thinks those options are syntax errors. As a result the theme doesn't load.
We would have to check if the `gtk+-2.0` version is smaller than `2.24.11` and if that is true the lines containing

    direction = LTR

or

    direction = RTL

have to be removed from the `gtkrc`.
I seriously doubt that it's worth the effort with Debian 8 just around the corner.
